internal/impl/generate.go

```diff
@@ -92,3 +98,36 @@ var templateFuncs = template.FuncMap{
 	"contains": strings.Contains,
 	"debug":    debug.IsEnabled,
 }
+
+func makeFlakeFile(outPath string, plan *plansdk.ShellPlan) error {
+
+	flakeDir := filepath.Join(outPath, "flake")
+	err := writeFromTemplate(flakeDir, plan, "flake.nix")
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	// make an empty git repo
+	// Alternatively consider: git add intent-to-add path/to/flake.nix, and
+	// git update-index --assume-unchanged path/to/flake.nix
+	// https://nixos.wiki/wiki/Flakes#How_to_add_a_file_locally_in_git_but_not_include_it_in_commits
+	cmd := exec.Command("git", "-C", flakeDir, "init")
```

Comment:
Why does the flake need to be added to git? What if the project isn't using git?

Reply:
to workaround this concern: https://github.com/NixOS/nix/issues/3121

That's okay, if the project is not using `git`, as long as their is a local `git` installed in the system.

Reply:
I'll revisit this. I think what we can do is:
- if in `git` repository, then do this step.
- if not in `git` repository, then can skip this step.